v0/destinations/marketo_bulk_upload/fileUpload.js

```diff
@@ -29,14 +35,53 @@ const getFileData = (input, config) => {
   let endTime;
   let requestTime;
   startTime = Date.now();
+  const headerArr = getHeaderFields(config);
+
+  // dedup starts
+  // Time Complexity = O(n2)
+  const dedupMap = new Map();
+  // iterating input and storing the occurences of messages
+  // with same dedup property received from config
+  // Example: dedup-property = email
+  // k (key)            v (index of occurence in input)
+  // user@email         [4,7,9]
+  // user2@email        [2,3]
+  // user3@email       [1]
+  input.map((i, index) => {
+    const indexAr = dedupMap.get(i.message[config.deDuplicationField]) || [];
+    indexAr.push(index);
+    dedupMap.set(i.message[config.deDuplicationField], indexAr);
+    return dedupMap;
+  });
+  // 1. iterating dedupMap
+  // 2. storing the duplicate occurences in dupValues arr
+  // 3. iterating dupValues arr, and mapping each property on firstBorn
+  // 4. as dupValues arr is sorted hence the firstBorn will inherit properties of last occurence (most updated one)
+  // 5. store firstBorn to first occurence in input as it should get the highest priority
+  dedupMap.forEach(indexes => {
+    const dupValues = [];
+    let firstBorn = {};
+    indexes.forEach(i => {
+      dupValues.push(input[i].message);
+    });
+    dupValues.forEach(dv => {
+      for (let i = 0; i < headerArr.length; i++) {
+        // if duplicate item has defined property to offer we take it else old one remains
+        firstBorn[headerArr[i]] = dv[headerArr[i]] || firstBorn[headerArr[i]];
+      }
+    });
+    firstBorn = removeUndefinedAndNullValues(firstBorn);
+    input[indexes[0]].message = firstBorn;
```

Comment:
While the first unique user/lead has all the merged information, the rest of the records are not removed from `input`

Reply:
Also, it is not a good idea to update the `input` itself.

Reply:
We are updating the `first occurence` with the upodated entry but keeping other entries as well as every entry has a job-id associated with them and those are updated in `successfulJobs` and `unsuccessfulJobs` arrays later. Hence we didn't want to make any changes with count of `input` as it might cause in-consistency with server jobs